contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,516 @@
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: []
+        action:
+    priorityClassNames:
+      - match: []
+        action:
+    tolerations:
+      - match: []
+        action:
+    nodeSelectors:
+      - match: []
+        action:
+    nodeAffinities:
+      - match: []
+        action:
+    podAntiAffinities:
+      - match: []
+        action:
+    podAffinities:
+      - match: []
+        action:
+```
+
+
+### required
+Elements here are required, pods won't schedule if they aren't present. Also note that if something is required it is also allowed.
+
+### allowed
+Elements here are allowed, the policy allows the presence of these elements. From Pod's perspective, a pod can use one or N of the allowed items.
+
+### deny
+If pods specify one of these, the pod won't schedule to a node, we won't dive into deny as it is the exact opposite of required.
+
+### Scheduler name
+
+If `schedulerNames` is absent from `allowed`, `default` or `required`, no scheduler is allowed by this specific policy.
+
+#### required
+
+Require that pods use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`):
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: ["green-scheduler","my-scheduler"]
+        action: required
+```
+
+An empty list of schedulerNames has no effect, as pod use the `defaultScheduler` if no scheduler is specified:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: []
+        action: required
+```
+
+
+#### allowed
+Allow pods to use either the green-scheduler (which is used by specifying `spec.schedulerName` in pod definition) or the `my-scheduler` scheduler (by specifying `spec.schedulerName: "my-scheduler"`) in the namespace `default`:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: ["green-scheduler","my-scheduler"]
+        action: allowed
+```
+
+An empty list of schedulerNames will allow usage of all schedulers:
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    schedulerNames:
+      - match: []
+        action: allowed
+```
+
+
+### Tolerations
+
+Toleration usage can be regulated using fine-grain rules with `tolerations` field. If specifying multiple `tolerations`, pod will be scheduled if one of the tolerations is satisfied.
```

Comment:
after reading scheduler names and tolerations, I feel it will be better to change `required` to be `require_any` or `require_one`, for less confusion.

Reply:
Since the matching behavior is the same for all action types, would you also want it to be `allow_any` and `deny_any`?